File: VB/SecuredExportExample.Module/Controllers/SecuredExportController.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Security
Imports DevExpress.ExpressApp.SystemModule
Imports SecuredExportExample.Module.SecurityObjects

Namespace SecuredExportExample.Module.Controllers
    Public Class SecuredExportController
        Inherits ViewController

        Protected Overrides Sub OnActivated()
            MyBase.OnActivated()
            Dim controller As ExportController = Frame.GetController(Of ExportController)()
            If controller IsNot Nothing Then
                AddHandler controller.ExportAction.Executing, AddressOf ExportAction_Executing
                If TypeOf SecuritySystem.Instance Is IRequestSecurity Then
                    controller.Active.SetItemValue("Security", SecuritySystem.IsGranted(New ExportPermissionRequest()))
                End If
            End If
        End Sub
        Private Sub ExportAction_Executing(ByVal sender As Object, ByVal e As System.ComponentModel.CancelEventArgs)
            SecuritySystem.Demand(New ExportPermissionRequest())
        End Sub
    End Class
End Namespace

File: CS/SecuredExportExample.Module/Controllers/SecuredExportController.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Security;
using DevExpress.ExpressApp.SystemModule;
using SecuredExportExample.Module.SecurityObjects;

namespace SecuredExportExample.Module.Controllers {
    public class SecuredExportController : ViewController {
        protected override void OnActivated() {
            base.OnActivated();
            ExportController controller = Frame.GetController<ExportController>();
            if (controller != null) {
                controller.ExportAction.Executing += ExportAction_Executing;
                if(SecuritySystem.Instance is IRequestSecurity) {
                    controller.Active.SetItemValue("Security", 
                        SecuritySystem.IsGranted(new ExportPermissionRequest()));
                }
            }
        }
        void ExportAction_Executing(object sender, System.ComponentModel.CancelEventArgs e) {
            SecuritySystem.Demand(new ExportPermissionRequest());
        }
    }
}
